### PR TITLE
Log bob deserialization exception for snapshot blobs

### PR DESCRIPTION
**Pull request by ajothomas**
**Symptom**: 
`BlobStoreUtil` doesn't log the blob id today when it fails to deserialize the snapshot blob. Without the blob id, it is tricky to debug why the deserialization issue happened.

**Changes**:
- Log the blob id if we are unable to deserialize the snapshot blobs

**Tests**:
- ./gradlew build

**API Changes**:
None

**Upgrade Instructions**: 
None

**Usage Instructions**: 
None

### CODE REVIEW

File: samza-core/src/main/java/org/apache/samza/storage/blobstore/util/BlobStoreUtil.java [hunk 1]
```diff
@@ -182,7 +182,13 @@ public CompletableFuture<SnapshotIndex> getSnapshotIndex(String blobId, Metadata
     return FutureUtil.executeAsyncWithRetries(opName, () -> {
       ByteArrayOutputStream indexBlobStream = new ByteArrayOutputStream(); // no need to close ByteArrayOutputStream
       return blobStoreManager.get(blobId, indexBlobStream, metadata, getDeleted).toCompletableFuture()
-          .thenApplyAsync(f -> snapshotIndexSerde.fromBytes(indexBlobStream.toByteArray()), executor);
+          .thenApplyAsync(f -> {
+            try {
+              return snapshotIndexSerde.fromBytes(indexBlobStream.toByteArray());
+            } catch (Exception e) {
+              throw new SamzaException(String.format("Unable to deserialize SnapshotIndex bytes for blob ID: %s", blobId), e);
```
Comment (dxichen) [inline samza-core/src/main/java/org/apache/samza/storage/blobstore/util/BlobStoreUtil.java:189]: could we use `.handle` for the exception instead?

Comment (ajothomas) [reply samza-core/src/main/java/org/apache/samza/storage/blobstore/util/BlobStoreUtil.java]: Done. thanks

**Review by dxichen — APPROVED** [review]

File: samza-core/src/main/java/org/apache/samza/storage/blobstore/util/BlobStoreUtil.java [hunk 2]
```diff
@@ -182,7 +182,13 @@ public CompletableFuture<SnapshotIndex> getSnapshotIndex(String blobId, Metadata
     return FutureUtil.executeAsyncWithRetries(opName, () -> {
       ByteArrayOutputStream indexBlobStream = new ByteArrayOutputStream(); // no need to close ByteArrayOutputStream
       return blobStoreManager.get(blobId, indexBlobStream, metadata, getDeleted).toCompletableFuture()
-          .thenApplyAsync(f -> snapshotIndexSerde.fromBytes(indexBlobStream.toByteArray()), executor);
+          .thenApplyAsync(f -> snapshotIndexSerde.fromBytes(indexBlobStream.toByteArray()), executor)
+          .handle((snapshotIndex, ex) -> {
+            if (ex != null) {
+              throw new SamzaException(String.format("Unable to deserialize SnapshotIndex bytes for blob ID: %s", blobId), ex);
```
Comment (xinyuiscool) [inline samza-core/src/main/java/org/apache/samza/storage/blobstore/util/BlobStoreUtil.java:188]: Seems we might get other kinds of exceptions here, e.g. GET exceptions, so change the log message to be "Unable to get SnapshotIndex ..." ?

**Review by xinyuiscool — COMMENTED** [review]
Could you please also check other places, e.g. GET a data blob, PUT for adding TTL, and see whether we log the blobId there when the exception happens? I think this change should apply across all the requests. Thanks.

**Pull request merged**